src/code/z_debug_display.c

```diff
@@ -1,15 +1,150 @@
+                                        GraphicsContext* gfxCtx) {
+    DebugDispObject* oldHead = sDebugObjectListHead;
+
+    sDebugObjectListHead = GRAPH_ALLOC(gfxCtx, sizeof(DebugDispObject));
+    sDebugObjectListHead->pos.x = posX;
+    sDebugObjectListHead->pos.y = posY;
+    sDebugObjectListHead->pos.z = posZ;
+    sDebugObjectListHead->rot.x = rotX;
+    sDebugObjectListHead->rot.y = rotY;
+    sDebugObjectListHead->rot.z = rotZ;
+    sDebugObjectListHead->scale.x = scaleX;
+    sDebugObjectListHead->scale.y = scaleY;
+    sDebugObjectListHead->scale.z = scaleZ;
+    sDebugObjectListHead->color.r = red;
+    sDebugObjectListHead->color.g = green;
+    sDebugObjectListHead->color.b = blue;
+    sDebugObjectListHead->color.a = alpha;
+    sDebugObjectListHead->type = type;
+    sDebugObjectListHead->next = oldHead;
+    return sDebugObjectListHead;
+}
+
+#include "code/debug_display/code.c"
+
+DebugDispObject_DrawFunc sDebugObjectDrawFuncTable[] = { DebugDisplay_DrawSpriteI8, DebugDisplay_DrawPolygon };
+
+DebugDispObjectInfo sDebugObjectInfoTable[] = {
+    { 0, sDebugDisplayCircleTex }, { 0, sDebugDisplayCrossTex }, { 0, sDebugDisplayBallTex },
+    { 0, sDebugDisplayCursorTex }, { 1, &sDebugDisplay1DL },     { 1, &sDebugDisplay3DL },
+    { 1, &sDebugDisplay2DL },
+};
+
+void DebugDisplay_DrawObjects(GlobalContext* globalCtx) {
+    DebugDispObject* dispObj = sDebugObjectListHead;
+    DebugDispObjectInfo* objInfo;
+
+    while (dispObj != NULL) {
+        objInfo = &sDebugObjectInfoTable[dispObj->type];
+        sDebugObjectDrawFuncTable[objInfo->drawType](dispObj, objInfo->drawArg, globalCtx);
+        dispObj = dispObj->next;
+    }
+}
+
+void DebugDisplay_DrawSpriteI8(DebugDispObject* dispObj, void* texture, GlobalContext* globalCtx) {
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C6FC(globalCtx->state.gfxCtx);
+
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, dispObj->color.r, dispObj->color.g, dispObj->color.b, dispObj->color.a);
+    Matrix_InsertTranslation(dispObj->pos.x, dispObj->pos.y, dispObj->pos.z, 0);
+    Matrix_Scale(dispObj->scale.x, dispObj->scale.y, dispObj->scale.z, 1);
+    Matrix_InsertMatrix(&globalCtx->mf_187FC, 1);
+    Matrix_InsertRotation(dispObj->rot.x, dispObj->rot.y, dispObj->rot.z, 1);
+    gDPSetTextureImage(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 1, texture);
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 0, 0x0000, G_TX_LOADTILE, 0, G_TX_NOMIRROR | G_TX_WRAP,
+               G_TX_NOMASK, G_TX_NOLOD, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOLOD);
+    gDPLoadSync(POLY_XLU_DISP++);
+
+    gDPLoadBlock(POLY_XLU_DISP++, G_TX_LOADTILE, 0, 0, 127, 1024);
+
+    gDPPipeSync(POLY_XLU_DISP++);
+
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_8b, 2, 0x0000, G_TX_RENDERTILE, 0, G_TX_NOMIRROR | G_TX_WRAP,
+               G_TX_NOMASK, G_TX_NOLOD, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOLOD);
+
+    gDPSetTileSize(POLY_XLU_DISP++, G_TX_RENDERTILE, 0, 0, 0x003C, 0x003C);
```

Comment:
If I'm not wrong, all of this is just simply this:
```suggestion
    gDPLoadTextureBlock(POLY_XLU_DISP++, texture, G_IM_FMT_I, G_IM_SIZ_8b, 16, 16, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
```